Makefile

```diff
@@ -366,13 +380,16 @@ generate-test-flavors: $(KUSTOMIZE)  ## Generate test template flavors
 e2e-image: docker-pull-prerequisites $(TOOLS_BIN_DIR)/start.sh $(TOOLS_BIN_DIR)/restart.sh ## Build an e2e test image
 	docker build -f Dockerfile --tag="gcr.io/k8s-staging-cluster-api/capa-manager:e2e" .
 
+$(SETUP_ENVTEST): # Build setup-envtest from tools folder.
+	GOBIN=$(abspath $(TOOLS_BIN_DIR)) $(GO_INSTALL) $(SETUP_ENVTEST_PKG) $(SETUP_ENVTEST_BIN) $(SETUP_ENVTEST_VER)
+
 .PHONY: test
-test: ## Run tests
-	source ./scripts/fetch_ext_bins.sh; fetch_tools; setup_envs; go test ./...
+test: $(SETUP_ENVTEST) ## Run tests
+	KUBEBUILDER_ASSETS="$(KUBEBUILDER_ASSETS)" go test -count 1 ./...
```

Comment:
Any reason to force not caching the tests as part of this change?

Reply:
I've got into this weird situation where I was running the tests over and over until I realised I'm not actually running them, because go is caching the result. :D And then I added this and it started to fail so I knew I was now running the right thing. :)

I can remove it of course.

Reply:
It would be good to remove the `-count 1` before merging.

We can always adopt the `$(TEST_ARGS)` like [upstream CAPI](https://github.com/kubernetes-sigs/cluster-api/blob/main/Makefile#L493:L499) to pass in additional args to go test in the future. wdyt?

Reply:
yeah, no worries. I'll remove it. :)

Reply:
Also, yes to TEST_ARGS. That sounds good.